### PR TITLE
[FEATURE] Panels can be viewed in full screen

### PR DESCRIPTION
**Pull request by Gladorme**
# Description

Add a panel action to show a panel "full-screen", it will still show nav + dashboard header (variables, ...)
Removing footer on dashboard view (better for full screen panel visualization) 🙈 

# Screenshots

Dashboard with no panel showed:
image

Dashboard with a panel showed:
image

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the
  following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `DOC`,`IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [x] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [x] Visual tests are stable and unlikely to be flaky.
  See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests)
  and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues
  include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

**Review by AntoineThebaud — APPROVED** [review]
LGTM

File: ui/dashboards/src/components/Panel/Panel.tsx [hunk 1]
```diff
@@ -100,6 +111,7 @@ export const Panel = memo(function Panel(props: PanelProps) {
         id={headerId}
         title={definition.spec.display.name}
         description={definition.spec.display.description}
+        readHandlers={readHandlers}
```
Comment (celian-garcia) [inline ui/dashboards/src/components/Panel/Panel.tsx:114]: somewhere I see view, somewhere I see full, here it says read.

I´d replace all wording by a common one no? What can we say fullView? focus? extended? maximized?

Comment (celian-garcia) [reply ui/dashboards/src/components/Panel/Panel.tsx:114]: Ok nevermind I think I follow what you're trying to do here.

**Review by celian-garcia — COMMENTED** [review]
That's really amazing! I love this new feature and the way you implemented it. 

I totally don't understand the zustand part and I guess it's been a nightmare to implement but you seem to not introduce new notion. Just subscribing yourself in the current implementation. 

Congrats!

**Comment by celian-garcia** [comment]
Little last comment, you could make the icon reversed when we're in full size and want to reduce back to normal.

File: ui/dashboards/src/components/Dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -26,16 +27,20 @@ export type DashboardProps = BoxProps & {
   emptyDashboardProps?: EmptyDashboardProps;
   panelOptions?: PanelOptions;
 };
+const HEADER_HEIGHT = 165; // Approximate height of the header in dashboard view (including the navbar and variables toolbar)
```
Comment (celian-garcia) [inline ui/dashboards/src/components/Dashboard/Dashboard.tsx:30]: Is this working if user zoom in the browser ? :smile:

File: ui/dashboards/src/context/DashboardProvider/DashboardProviderWithQueryParams.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { StringParam, useQueryParam } from 'use-query-params';
+import { DashboardProvider, DashboardProviderProps } from './DashboardProvider';
+
+export function DashboardProviderWithQueryParams({ children, initialState }: DashboardProviderProps) {
```
Comment (celian-garcia) [inline ui/dashboards/src/context/DashboardProvider/DashboardProviderWithQueryParams.tsx:17]: <3

**Review by celian-garcia — APPROVED** [review]

File: ui/dashboards/src/components/Dashboard/Dashboard.tsx (hunk 2 above)
Comment (Gladorme) [reply ui/dashboards/src/components/Dashboard/Dashboard.tsx:30]: Well it's an approximate height for desktop screen, this value is here only while the ref is not loaded. I think I could even put 0 🤔

**Pull request merged**